### PR TITLE
Remove munin

### PR DESCRIPTION
**Pull request by Firefishy**
Remove munin, but initially leave munin includes for cleanup.

Fixed: https://github.com/openstreetmap/operations/issues/501

### CODE REVIEW

File: cookbooks/mysql/recipes/default.rb [hunk 1]
```diff
@@ -55,31 +55,10 @@
   only_if { ::Dir.exist?("/sys/kernel/security/apparmor") }
 end
 
+# FIXME: These packages only used for former munin plugins?
```
Comment (tomhughes) [inline cookbooks/mysql/recipes/default.rb:58]: Probably, yes, so they can be purged?

File: cookbooks/nginx/recipes/default.rb [hunk 2]
```diff
@@ -53,15 +53,8 @@
   subscribes :restart, "template[/etc/nginx/nginx.conf]"
 end
 
-munin_plugin_conf "nginx" do
-  template "munin.erb"
-end
-
 package "libwww-perl"
```
Comment (tomhughes) [inline cookbooks/nginx/recipes/default.rb:56]: Suspect this is also only used by the plugin?

File: cookbooks/overpass/recipes/default.rb [hunk 3]
```diff
@@ -27,7 +27,7 @@
 basedir = data_bag_item("accounts", username)["home"]
 web_passwords = data_bag_item("web", "passwords")
 
-%w[bin site diffs db src munin].each do |dirname|
+%w[bin site diffs db src].each do |dirname|
```
Comment (tomhughes) [inline cookbooks/overpass/recipes/default.rb:30]: We probably want to delete the munin directory?

File: cookbooks/db/recipes/master.rb [hunk 4]
```diff
@@ -62,10 +62,11 @@
   password passwords["backup"]
 end
 
-postgresql_user "munin" do
-  cluster node[:db][:cluster]
-  password passwords["munin"]
-end
+# FIXME: munin user should be removed
+# postgresql_user "munin" do
```
[the remark this answers is not in the document]
Comment (Firefishy) [reply cookbooks/db/recipes/master.rb]: Switched to `action :drop`

File: cookbooks/munin/recipes/default.rb [hunk 5]
```diff
@@ -17,17 +17,14 @@
...
-  action [:enable, :start]
-  supports :status => true, :restart => true, :reload => true
+  action [:stop, :disable]
 end
 
 servers = search(:node, "recipes:munin\\:\\:server").map(&:ipaddresses).flatten
 
 firewall_rule "accept-munin" do
-  action :accept
+  action :reject
```
[the remark this answers is not in the document]
Comment (Firefishy) [reply cookbooks/munin/recipes/default.rb]: Removed the resource.

File: cookbooks/munin/recipes/default.rb [hunk 6]
```diff
@@ -36,11 +33,10 @@
...
-  group "root"
-  mode "644"
-  variables :servers => servers
-  notifies :restart, "service[munin-node]"
+file "/etc/munin/munin-node.conf" do
+  action :delete
+end
+
+package "munin-node" do
+  action :purge
```
[the remark this answers is not in the document]
Comment (Firefishy) [reply cookbooks/munin/recipes/default.rb:29]: role `base` pulls in `recipe[munin::plugins]` which is used for purging `/etc/munin/plugins`, `/usr/local/share/munin/plugins` and `/etc/munin/plugin-conf.d` directories.

I suspect `/etc/munin` directory may need to be purged by a follow-up action.

File: cookbooks/munin/recipes/server.rb [hunk 7]
```diff
@@ -37,72 +30,54 @@
...
-remote_directory "/srv/munin.openstreetmap.org" do
-  source "www"
-  owner "root"
-  group "root"
-  mode "755"
-  files_owner "root"
-  files_group "root"
-  files_mode "644"
-end
+# remote_directory "/srv/munin.openstreetmap.org" do
```
[the remark this answers is not in the document]
Comment (Firefishy) [reply cookbooks/munin/recipes/server.rb]: Now deleted.

File: cookbooks/munin/recipes/server.rb [hunk 8]
```diff
@@ -37,72 +30,54 @@
...
+end
+
+package "rrdcached" do
+  action :purge
 end
 
-munin_plugin "munin_stats"
-munin_plugin "munin_update"
-munin_plugin "munin_rrdcached"
+# package "libcgi-fast-perl" do
```
[the remark this answers is not in the document]
Comment (Firefishy) [reply cookbooks/munin/recipes/server.rb]: Purged now. I was paranoid we used it elsewhere.

File: cookbooks/mysql/recipes/default.rb (hunk 1 above)
Comment (Firefishy) [reply cookbooks/mysql/recipes/default.rb]: Purged now.

File: cookbooks/overpass/recipes/default.rb (hunk 3 above)
Comment (Firefishy) [reply cookbooks/overpass/recipes/default.rb:30]: Purged now.

**Review by tomhughes — APPROVED** [review]

**Pull request merged**